test-tosam.sh

```diff
@@ -4,6 +4,8 @@ success=1
 for f in test-data/*.mas; do
     if ! cmp -s <(./tosam $f) `basename $f .mas`.sam; then
         echo "./tosam $f did not produce the expected output"
+        diff <(./tosam $f) test-data/*.sam
```

Comment:
This is a bit dodgy. You are trying to diff between a single file and all files in `test-data`. That will generally not work, although it might when there is only one... I am not sure. It is not what you want, though. You want to compare `<(./tosam $f)` with `` test-data/`basename $f .mas`.sam `` like the test above.